### PR TITLE
fix(verifiers): reject untrusted audit packet evidence

### PR DESCRIPTION
**Pull request by luckyPipewrench**
## Summary

- require nonempty chains and external trust anchors
- harden pinned-key parsing, packet metadata, and hash-pinned reads
- add cross-language regression coverage and an upgrade note

## Why

Verifier edge cases could promote empty or attacker-controlled evidence.

## Summary by CodeRabbit

* **Breaking Changes / Upgrade Notes**
  * Trusted Audit Packet verification now fails closed unless an external trust anchor is supplied via `--key` or `--expect-sha256` (embedded keys alone no longer establish provenance).
* **Bug Fixes**
  * Empty receipt chains are rejected.
  * `final_seq` is now cross-checked based on explicit presence (including `final_seq: 0`).
  * Inline signer-key literals take precedence over same-named key files; empty external signer keys are rejected.
* **Documentation**
  * Updated verification examples and verifier READMEs to reflect required trust-anchor input and verification modes.
* **Tests**
  * Expanded Rust and TypeScript coverage for trust/anchoring, empty-chain handling, and `final_seq` edge cases.

### CODE REVIEW

File: sdk/verifiers/ts/src/audit-packet.ts [hunk 1]
```diff
@@ -94,7 +94,7 @@ function crossCheck(packet: AuditPacket, chain: ChainResult, receipts: Receipt[]
   if (packet.verifier?.root_hash && packet.verifier.root_hash !== chain.root_hash) {
     errors.push(`root_hash mismatch: chain=${chain.root_hash} packet=${packet.verifier.root_hash}`);
   }
-  if ((packet.verifier?.final_seq ?? 0) !== 0 && packet.verifier?.final_seq !== chain.final_seq) {
+  if (packet.verifier?.final_seq !== undefined && packet.verifier.final_seq !== chain.final_seq) {
```
[the remark this answers is not in the document]
Comment (luckyPipewrench) [reply sdk/verifiers/ts/src/audit-packet.ts:97]: `final_seq` remains optional because Audit Packet v0 requires only `verdict` and `trusted`; omission means no sequence claim. Go and Rust use the same presence-aware cross-check, while explicit zero and non-zero values remain validated. Requiring it only in TypeScript would break cross-verifier compatibility, and requiring it everywhere would be a v0 compatibility change.

**Review by DeathCamel58 — APPROVED** [review]
